scripts/west_commands/runners/nrfjprog.py

```diff
@@ -111,7 +111,10 @@ def get_board_snr(self, glob):
         p = 'Please select one with desired serial number (1-{}): '.format(
                 len(snrs))
         while True:
-            value = input(p)
+            try:
+                value = input(p)
+            except EOFError:
+                sys.exit(0)
```

Comment:
This only works if `CTRL-D` is pressed when waiting for the first character.

In this case:
```
Please select one with desired serial number (1-2): 1<CTRL>-D
```
then nothing happens.

this works:
```
Please select one with desired serial number (1-2): <CTRL>-D
```

Reply:
@mbolivar-nordic  you might want to address this too.

Reply:
The input() function is part of the python standard library. It doesn't return control to the user on the first control-d if characters have already been typed.

You have to keep hitting control-D multiple times to get it to exit. Then you can hit control-D again when it reprompts, like this:

```
$ west flash
[...]
There are multiple boards connected.
[...]
Please select one with desired serial number (1-2): xxx^D^D^DPlease select one with desired serial number (1-2): ^D
$
```

If you've already entered some characters, control-C already works. I'm just adding control-D support in this PR.